src/main.ts

```diff
@@ -70,25 +71,38 @@ async function run(): Promise<void> {
         const installationPath = data.toString().trim()
         core.debug(`Found installation path: ${installationPath}`)
 
-        let toolPath = path.join(
-          installationPath,
-          'MSBuild\\Current\\Bin\\MSBuild.exe'
-        )
-
-        core.debug(`Checking for path: ${toolPath}`)
-        if (!fs.existsSync(toolPath)) {
-          toolPath = path.join(
+        // x64 only exists in one possible location, so no fallback probing
+        if(MSBUILD_ARCH === "x64") {
```

Comment:
Nit for consistency:
```suggestion
        if (MSBUILD_ARCH === "x64") {
```